flowmachine/flowmachine/core/query.py

```diff
@@ -498,63 +491,43 @@ def to_sql_async(
             ).format(name, len(name), MAX_POSTGRES_NAME_LENGTH)
             raise NameTooLongError(err_msg)
 
-        def do_query():
+        def do_query() -> Query:
             logger.debug("Getting storage lock.")
             with rlock(self.redis, self.md5):
                 logger.debug("Obtained storage lock.")
-                Qs = self._make_sql(name, schema=schema, as_view=as_view, force=force)
+                Qs = self._make_sql(name, schema=schema, force=force)
                 logger.debug("Made SQL.")
                 con = self.connection.engine
-                if force and not as_view:
+                if force:
                     self.invalidate_db_cache(name, schema=schema)
+                plan_time = 0
                 with con.begin():
-                    sql = """CREATE SCHEMA IF NOT EXISTS {}""".format(schema)
-                    con.execute(sql)
-                    try:
-                        for Q in Qs:
-                            con.execute(Q)
-                    except Exception as e:
-                        logger.error(f"Error executing SQL: '{Q}'. Error was {e}")
-                        raise e
+                    rs = []
```

Comment:
I'll be honest that this loop does read a little cryptic at first. Maybe it's due to the short names `Qs`, `rs` and the fact that it's not immediately clear what types of objects we're dealing with (e.g. what type of object is `r` and why is `r[0][0][0]` a query plan(?) and how do we know that it has a key called `"Execution Time"`?

Not sure I have any good suggestions other than maybe slightly more explicit names?

Also, why is `Qs` an iterable? Shouldn't `self._make_sql(...)` just return the SQL string?

Reply:
Given these slightly less terse names.

Qs (as was) is a list of ddl ops. Marginally nicer to have this as an iterable because you can report the specific one that failed.

r[0][0][0] should be a (json) query plan which Postgres helpfully returns as a row, with one field, and wrapped in a list.